src/pages/experiments/[experimentId]/plots-and-tables/marker-heatmap/index.jsx

```diff
@@ -214,106 +209,37 @@ const MarkerHeatmap = ({ experimentId }) => {
     }
 
     const data = populateHeatmapData(cellSets, config, expressionData, config.selectedGenes, true);
+    const spec = generateSpec(config, 'Cluster ID', data, true);
 
-    const spec = generateSpec(config, 'Cluster ID', data.trackGroupData, plotUuid);
+    spec.description = 'Marker heatmap';
 
-    const newVegaSpec = {
-      ...spec,
-      axes: [...spec.axes, ...displayLabels()],
-      data: spec.data.map((datum) => ({
-        ...datum,
-        values: data[datum.name],
-      })),
+    const extraMarks = {
```

Comment:
I feel like this stuff should happen inside the generateSpec not here? Also in the other heatmap i saw the same thing for the axes

Reply:
The generateHeatmapSpec functionality is used for both the marker heatmap and the heatmap plot. What differentiates them is just these markings. I don't think passing an additional param to the `generateHeatmapSpec` is a good way to differentiate between the two. That's why I take them out and put them here.

What I can do to make it look better is maybe to wrap it inside a wrapper function which adds these settings to the output of `generateHeatmapSpec`?

Reply:
I think that just complicates things more. Maybe just keep it like this?

Reply:
I'll create the wrapper function to make it easier to read.